crates/services/sync/src/import.rs

```diff
@@ -0,0 +1,331 @@
+//! # Importer Task
+//! This module contains the import task which is responsible for
+//! importing blocks from the network into the local blockchain.
+
+use std::{
+    ops::RangeInclusive,
+    sync::Arc,
+};
+
+use fuel_core_services::{
+    SharedMutex,
+    Shutdown,
+    SourcePeer,
+};
+use fuel_core_types::blockchain::{
+    block::Block,
+    consensus::Sealed,
+    primitives::BlockHeight,
+    SealedBlock,
+    SealedBlockHeader,
+};
+use futures::{
+    stream::{
+        self,
+        StreamExt,
+    },
+    Stream,
+};
+use std::future::Future;
+use tokio::sync::Notify;
+
+use crate::{
+    ports::{
+        ConsensusPort,
+        ExecutorPort,
+        PeerToPeerPort,
+        Ports,
+    },
+    state::State,
+};
+
+#[cfg(test)]
+pub(crate) use tests::empty_header;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(test)]
+mod back_pressure_tests;
+
+#[derive(Clone, Copy, Debug)]
+/// Parameters for the import task.
+pub struct Params {
+    /// The maximum number of get header requests to make in a single batch.
+    pub max_get_header_requests: usize,
+    /// The maximum number of get transaction requests to make in a single batch.
+    pub max_get_txns_requests: usize,
+}
+
+/// Long running task that imports blocks from the network and executes them.
+pub(super) async fn import<P, E, C>(
+    state: SharedMutex<State>,
+    notify: Arc<Notify>,
+    params: Params,
+    ports: Ports<P, E, C>,
+    shutdown: Shutdown,
+    // Needed for testing. Would be nice to find a better way to do this.
```

Comment:
If you had the `run` function with one iteration of the importer service, you could easily get the result after its execution and compare them ;)

Reply:
But then there's no way to wait on the notify.

Reply:
First, you can send the notification and after, you can do `run.await`

Reply:
I don't understand?